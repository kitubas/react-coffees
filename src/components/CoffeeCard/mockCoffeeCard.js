export const coffee = {
    title: 'title 1',
    description: 'body 1',
    image: 'img/img.png'
  }