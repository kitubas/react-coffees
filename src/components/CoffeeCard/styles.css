.coffee {
    background: #fff;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
    transition: transform 100ms ease-in-out;
    border: #000;
    border-style: solid;
    padding: 0.3em;
  }
  
  .coffee:hover {
    transform: scale(1.05);
  }
  
  .coffee img {
    max-width: 100%;
  }