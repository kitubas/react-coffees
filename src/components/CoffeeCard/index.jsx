import P from 'prop-types';
import './styles.css';

export const CoffeeCard = ({ coffee }) => (
  <div className="coffee">
    <img src={coffee.image} alt={coffee.title} />
    <div className="coffee-content">
      <h2>{coffee.title}</h2>
      <p>{coffee.description}</p>
    </div>
  </div>
);

CoffeeCard.propTypes = {
  coffee: P.object.isRequired,
};
