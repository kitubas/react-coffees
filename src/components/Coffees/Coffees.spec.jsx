import { render, screen } from "@testing-library/react";
import { Coffees } from ".";
import { coffee as coffeeMock } from "../CoffeeCard/mockCoffeeCard";

const props = {
  coffees: [coffeeMock],
};

describe("<Coffees />", () => {
  it('should render Coffees', () => {
    render(<Coffees {...props} />);

    expect(screen.getAllByRole("heading", { name: /title/i })).toHaveLength(1);
    expect(screen.getAllByRole("img", { name: /title/i })).toHaveLength(1);
    expect(screen.getAllByText(/body/i)).toHaveLength(1);
    expect(screen.getByRole("img", { name: /title 1/i })).toHaveAttribute(
      "src",
      "img/img.png"
    );
  });

  it('should not render Coffees', () => {
    render(<Coffees />);
    expect(screen.queryByRole('heading', { name: /title/i }))
      .not.toBeInTheDocument();
  });
  
  it('should match snapshot', () => {
    const {container} = render(<Coffees {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
