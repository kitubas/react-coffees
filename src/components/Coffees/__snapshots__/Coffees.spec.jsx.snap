// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Coffees /> should match snapshot 1`] = `
<div
  class="coffees"
>
  <div
    class="coffee"
  >
    <img
      alt="title 1"
      src="img/img.png"
    />
    <div
      class="coffee-content"
    >
      <h2>
        title 1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>
</div>
`;
